includes/wf_crm_webform_postprocess.inc

```diff
@@ -1102,6 +1102,22 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
       $params['skipStatusCal'] = !empty($params['status_id']);
 
       $result = wf_civicrm_api('membership', 'create', $params);
+      
+      //Issue #2516924 If existing membership create renewal activity
+      foreach ($result['values'] as $memId => $membership) {
+        if ($params['membership_type_id'] == $memId) {
```

Comment:
It looks like you're comparing apples to oranges here. Why would it matter if the membership type matched the membership id?